### DISCUSSION TITLE
What is the point of IP ranges if they don't also have a related IP address?

### DISCUSSION THREAD

**Discussion by Mailstorm-ctrl** [netbox-community/netbox#16087] [Q&A]
For example, let's say we have a DHCP range 10.10.100.10/24 to 10.10.100.100/24.

If we make an IP range of that and mark it as:
- Reserved
- Fully utilized

How come I can make a single IP in that range that can have a different status? If I was dynamically making hosts (that have static addresses) and the next available IP reported as available was 10.10.100.11, we'd have a problem. The host is going to have that IP statically assigned but unless you are specifically looking at ranges, you wouldn't know that IP is in the DHCP scope defined with an IP range.

Now yes, the DHCP server should first see if any device on the network has that IP assigned already and not give it, but that's not the point. The point is what you have documented/want to happen isn't enforced.

I'd expect if I made a range like above with those conditions (or similar), then those IPs in that range should also be made so they aren't "available".

Maybe I'm overthinking this but IP ranges in their current state don't make sense to me.

**Comment by samip5** [comment]
Yes, this doesn't make sense. You could have reserved the whole prefix in prefix-range, but next-available would still pick one that could conflict.

**Reply by markkuleinio** [reply]
Can you be more specific about your problem? Here I don't get conflicting allocations (with `available-ips`) when **Mark populated** has been set for the IP range.

**Reply by samip5** [reply]
> Can you be more specific about your problem? Here I don't get conflicting allocations (with `available-ips`) when **Mark populated** has been set for the IP range.

I may not want to set it as mark-populated as that will prevent me from adding manually IPs inside the range.
Eg I may want to have range of DHCP prefixes but I may want to assign static addresses inside the same range if those are served by DHCP.

**Reply by jeremystretch** [reply]
You are replying to a post that is nearly two years old. A lot has changed in the application since then. If you have a question about how NetBox works, please start a new thread instead of replying to an outdated one.

**Reply by samip5** [reply]
> You are replying to a post that is nearly two years old. A lot has changed in the application since then. If you have a question about how NetBox works, please start a new thread instead of replying to an outdated one.

My problem exists in 4.x and as this was one of the google hits, it made sense to reply to this one.

**Reply by jeremystretch** [reply]
There are broad functional differences between each minor release of NetBox, as each represents four months of development effort. That this thread was the first search result does not indicate it is actually relevant to the behavior you're seeing today.
